### PR TITLE
Scalafix rules for Monocle 3 Resolves Issue #1001

### PR DESCRIPTION
**Pull request by dsebban**
- [x] Set up scalafix infra
- [x] Make it work for set -> replace
- [ ] Add more rules
- [ ]  Do I need more advanced rules ? like inside map and flatMap ?

### CODE REVIEW

File: scalafix/rules/src/main/scala/fix/Monocle3.scala [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+class Monocle3 extends SemanticRule("Monocle") {
+
+  override def fix(implicit doc: SemanticDocument): Patch = {
+    SetRules(doc.tree)
+  }.asPatch
+
+  object SetRules {
+    private[this] val setMatcher =
+      SymbolMatcher.normalized("monocle/PLens#set.")
```
Comment (julien-truffaut) [inline scalafix/rules/src/main/scala/fix/Monocle3.scala:14]: How can we add other optics type? `PPrism`, `POptional`, `PTraversal` and `PSetter`

File: scalafix/build.sbt [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+  libraryDependencies ++= Seq(
+    "com.github.julien-truffaut" %% "monocle-core" % "2.0.3"
+  )
+)
+
+lazy val output = project.settings(
+  skip in publish := true,
+  skip in compile := true,
+  libraryDependencies ++= Seq(
+    "com.github.julien-truffaut" % "monocle-core_3.0.0-M1" % "3.0.0-M0a"
```
Comment (julien-truffaut) [inline scalafix/build.sbt:39]: how does it work? is the middle string the target version?

File: scalafix/rules/src/main/scala/fix/Monocle3.scala (hunk 1 above)
Comment (dsebban) [reply scalafix/rules/src/main/scala/fix/Monocle3.scala:14]: Yes will get to it this week hopefully

**Comment by dsebban** [comment]
Sorry, I don't have time to work on this anymore, I will close the PR in order for someone else to take it

**Pull request closed without merging**